docs/source/configuration_file.md

```diff
@@ -0,0 +1,159 @@
+
+#### scheduler
+- `initial_lr`: Float. Initial learning rate.
+- `scheduler_lr`:
+1. `name`: Choice between: `"CosineAnnealingLR"`, `"CosineAnnealingWarmRestarts"` and `"CyclicLR"`. Please find documentation [here](https://pytorch.org/docs/stable/optim.html).
+2. Other parameters that are needed for the scheduler of interest (e.g. `"base_lr": 1e-5, "max_lr": 1e-2` for `"CosineAnnealingLR"`).
+
+#### balance_samples
+Bool. Balance positive and negative labels in both the training and the validation datasets.
+
+#### mixup_alpha
+Float. Alpha parameter of the Beta distribution, see [original paper](https://arxiv.org/abs/1710.09412).
+
+#### transfer_learning
+- `retrain_model`: Filename of the pretrained model (`path/to/pretrained-model`). If `null`, no transfer learning is performed and the network is trained from scratch.
+- `retrain_fraction`: Float between 0. and 1. Controls the fraction of the pre-trained model that will be fine-tuned. For instance, if set to 0.5, then the second half of the model will be fine-tuned, while the first layers will be frozen.
+
+## Architecture
+Architectures for both segmentation and classification are available and described [here](models.rst).
+If the selected architecture is listed [here](../../ivadomed/loader/loader.py#L14), then a classification task is run, ie the ground-truth are labels extracted from `target`, instead of arrays for the segmentation task.
+
+### default_model (Mandatory)
+Define the default model (`Unet`) and mandatory parameters that are common to all available architectures (listed [here](models.rst)). If a tailored model is defined (see next section), then the default parameters are merged with the parameters that are specific to the tailored model.
+- `name`: `Unet` (default)
+- `dropout_rate`: float (e.g. 0.4).
+- `batch_norm_momentum`: float (e.g. 0.1).
+- `depth`: int, number of down-sampling operations.
+Note:
+- `in_channel` is automatically defined with `multichannel` and `contrast/training_validation`.
+- `out_channel` is automatically defined with `target_suffix`.
+
+### Tailored model (optional)
+Here are defined the tailored model and the parameters that are specific to it (ie not defined in the default model). See examples:
+- [FiLMedUnet](../../ivadomed/config/config.json#L64)
+    - `metadata`: choice between `"without"`, `"mri_params"`, and `"contrast"`. If `"mri_params"`, then vectors of [FlipAngle, EchoTime, RepetitionTime, Manufacturer] are input to the FiLM generator. If `"contrast"`, then image contrasts (according to `config/contrast_dct.json`) are input to the FiLM generator.
+- [HeMISUnet](../../ivadomed/config/config_spineGeHemis.json#L64)
+    - `missing_contrast`: Bool.
+- [UNet3D](../../ivadomed/config/config_tumorSeg.json#L65)
+    - `length_3D`: tuple indicating the size of the subvolumes or volume used for unet 3D model (depth, width, height).
```

Comment:
depth, width and height might be confusing: do they correspond to the first, second and third dimension of a numpy array, respectively? If so i would use that terminology instead (no confusion possible)

Reply:
What about:
```
- `length_3D`: (Integer, Integer, Integer). Size of the 3D patches used as model's input tensors, along the first, second and third dimension.
```

Reply:
i think you can even go with: "- `length_3D`: (Integer, Integer, Integer). Size of the 3D patches used as model's input tensors.".  It is fairly self-explanatory that the 1st, 2nd and 3rd arguments refer to the 1st, 2nd and 3rd dimension, although it was not clear before when the terms width, depth, etc. were used (because these terms imply a certain convention on the object itself)